test/avocado/testlib_avocado/seleniumlib.py

```diff
@@ -105,39 +110,50 @@ def connect_browser():
 
         @Retry(attempts=3, timeout=30,
                exceptions=(WebDriverException,),
-               error=Exception('Timeout: Unable to get page'))
+               error=SeleniumFailure('Timeout: Unable to get page'))
         def connectwebpage():
             self.driver.get('%s://%s:%s' % (url_base, guest_machine, network_port))
 
         connectwebpage()
 
-        # if self.error evaluates to True when a test finishes,
-        # an error is raised and a screenshot generated
-        self.error = True
+        self.error_setup = False
 
-    def tearDown(self):
-        if self.error:
-            screenshot_file = ""
+    def _get_screenshot_name(self, *args):
+        sep="-"
+        if len(args) > 0:
+            suffix = sep.join(args)
+        else:
+            datesuffix = str(time.clock())[2:]
             try:
-                # use time.clock() to ensure that snapshot files are unique and ordered
-                # sample name is like: screenshot-teardown-172434.png
-                screenshot_file = "screenshotTeardown%s.png" % str(time.clock())[2:]
-
-                self.driver.save_screenshot(os.path.join(actualpath,screenshot_file))
-                self.log.error("Screenshot(teardown) - Wrote: " + screenshot_file)
+                stackinfo = [x[3] for x in inspect.stack() if x[3].startswith("test") or x[3] in ["tearDown", "setUp"]]
+            except Exception:
```

Comment:
This is a blanket exception handler again, which would also catch things like `AttributeError` or `NameError`. As the point of this PR is to clean up error handling, let's please not introduce new ones.

Reply:
@martinpitt I understand, but the trouble is that I were unable to raise any exception. It is somehow connected that stack info is missing, but i have no idea if it raise any exception. So I can use there this NameError and attribute one for sure, seems these could be good ones.

Reply:
I'm afraid I don't understand this. But then again I also don't understand what kind of Exception you would possibly expect here in the first place. E. g. you *never* want to intercept `AttributeError` or `NameError`, these are usually typos in your program that blanket exception catching hides. It is possible to have a `KeyError` if you expect these arrey indexes to sometimes not exist, or `inspect.stack()` might fail.

Did you try what happens if you don't catch exceptions here at all?

Reply:
I've looked to doc and it seems that there is no exception raised, just Null if not supported. so I'll repair it